### ISSUE TITLE
Lint based on file type 

### ISSUE THREAD

**Issue by lekkas** [resin-io/node-resin-lint#3]

**Comment by thgreasi** [comment]
@lekkas does this implies that it should also handle Typescript?
Should we PR the UI's tslint configuration into this?
See: https://github.com/resin-io/resin-ui/issues/1418

**Comment by lekkas** [comment]
@thgreasi yeah, the idea behind `node-resin-lint` was to abstract more linters per file/language type with the benefit of having their linting configuration in this repo. It'd be great if we had this for Typescript as well!
